src/main/host/syscall/handler/timerfd.rs

```diff
@@ -0,0 +1,244 @@
+            return Err(Errno::EINVAL.into());
+        };
+
+        Self::timerfd_gettime_helper(ctx, timerfd, curr_value_ptr)?;
+
+        Ok(0.into())
+    }
+
+    fn timerfd_gettime_helper(
+        ctx: &mut SyscallContext,
+        timerfd: &Arc<AtomicRefCell<TimerFd>>,
+        value_ptr: ForeignPtr<libc::itimerspec>,
+    ) -> Result<(), Errno> {
+        // Lookup the timer state.
+        let (remaining, interval) = {
+            let borrowed = timerfd.borrow();
+
+            // We return a zero duration if the timer is disarmed.
+            let remaining = match borrowed.get_timer_remaining() {
+                Some(simtime) => simtime,
+                None => SimulationTime::ZERO,
+            };
+            // We return a zero duration if the timer is non-periodic (set to expire only once).
+            let interval = borrowed.get_timer_interval();
+
+            (remaining, interval)
+        };
+
+        // Write the result to the plugin.
+        {
+            // Need a location to write the result.
+            if value_ptr.is_null() {
+                return Err(Errno::EFAULT);
+            }
+
+            // Error if the process disappeared.
+            let Some(runnable) = ctx.objs.process.borrow_runnable() else {
+                return Err(Errno::EINVAL);
+            };
+
+            // Set up the result values.
+            let result = libc::itimerspec {
+                it_value: remaining.try_into().unwrap(),
+                it_interval: interval.try_into().unwrap(),
+            };
+
+            // Write the result.
+            runnable.memory_borrow_mut().write(value_ptr, &result)?;
+        }
+
+        Ok(())
+    }
+
+    #[log_syscall(/* rv */ libc::c_int, /* fd */ libc::c_int, /* flags */ libc::c_int,
+                  /* new_value */ *const libc::c_void, /* old_value */ *const libc::c_void)]
+    pub fn timerfd_settime(
+        ctx: &mut SyscallContext,
+        fd: libc::c_int,
+        flags: libc::c_int,
+        new_value_ptr: ForeignPtr<libc::itimerspec>,
+        old_value_ptr: ForeignPtr<libc::itimerspec>,
+    ) -> SyscallResult {
+        let Some(flags) = TimerSetTimeFlags::from_bits(flags) else {
+            log::warn!("Invalid timerfd_settime flags: {}", flags);
+            return Err(Errno::EINVAL.into());
+        };
+
+        // Get the TimerFd object.
+        let file = get_cloned_file(ctx, fd)?;
+        let File::TimerFd(ref timerfd) = file else {
+            return Err(Errno::EINVAL.into());
+        };
+
+        // Read in the new value from the plugin.
+        let new_value = {
+            // Need to be able to read the new timer config.
+            if new_value_ptr.is_null() {
+                return Err(Errno::EFAULT.into());
+            }
+
+            // Error if the process disappeared.
+            let Some(runnable) = ctx.objs.process.borrow_runnable() else {
+                return Err(Errno::EINVAL.into());
+            };
```

Comment:
Same as comment above. Can we just do `ctx.objs.process.memory_borrow().read(new_value_ptr)` later without getting this `runnable`)?